src/thinblock.cpp

```diff
@@ -984,6 +984,93 @@ double CThinBlockData::average(std::map<int64_t, uint64_t> &map)
     return (double)accum / map.size();
 }
 
+double CThinBlockData::computTotalBandwidthSavingsInternal() EXCLUSIVE_LOCKS_REQUIRED(cs_thinblockstats)
+{
+    AssertLockHeld(cs_thinblockstats);
+
+    return double(nOriginalSize() - nThinSize() - nTotalBloomFilterBytes());
+}
+
+double CThinBlockData::compute24hInboundCompressionInternal() EXCLUSIVE_LOCKS_REQUIRED(cs_thinblockstats)
+{
+    AssertLockHeld(cs_thinblockstats);
+
+    expireStats(mapThinBlocksInBound);
+
+    double nCompressionRate = 0;
+    uint64_t nThinSizeTotal = 0;
+    uint64_t nOriginalSizeTotal = 0;
+    for (std::map<int64_t, std::pair<uint64_t, uint64_t> >::iterator mi = mapThinBlocksInBound.begin();
+         mi != mapThinBlocksInBound.end(); ++mi)
+    {
+        nThinSizeTotal += (*mi).second.first;
+        nOriginalSizeTotal += (*mi).second.second;
+    }
+    // We count up the outbound bloom filters. Outbound bloom filters go with Inbound xthins.
+    uint64_t nOutBoundBloomFilterSize = 0;
+    for (std::map<int64_t, uint64_t>::iterator mi = mapBloomFiltersOutBound.begin();
+         mi != mapBloomFiltersOutBound.end(); ++mi)
+    {
+        nOutBoundBloomFilterSize += (*mi).second;
+    }
+
+    if (nOriginalSizeTotal > 0)
+        nCompressionRate = 100 - (100 * (double)(nThinSizeTotal + nOutBoundBloomFilterSize) / nOriginalSizeTotal);
+
+    return nCompressionRate;
+}
+
+double CThinBlockData::compute24hOutboundCompressionInternal() EXCLUSIVE_LOCKS_REQUIRED(cs_thinblockstats)
```

Comment:
Hmm, this is an almost repeat of `compute24hInboundCompressionInternal()`, isn't it?

Reply:
Will make the same DRY updates as requested for graphene.

Reply:
Fixed

Reply:
Great, other change as well!